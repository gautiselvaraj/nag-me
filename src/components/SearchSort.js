import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';
import Search from './Search';
import Select from './Select';

const SearchSortWrap = styled.div`
  align-items: center;
  background-color: ${props=> props.theme.white};
  border-bottom: 1px solid ${props=> props.theme.greyLighter};
  display: flex;
  padding: 5px;
  justify-content: space-between;
`;

const SearchWrap = styled.div`
  flex-grow: 1;
`;

const SortWrap = styled.div`
  color: ${props => props.theme.greyDark};
  margin-left: 10px;
`;

let SearchSort = ({nagsSearch, nagsSort}) => (
  <SearchSortWrap>
    <SearchWrap>
      <Search handleKeyup={e => nagsSearch(e.target.value)} />
    </SearchWrap>
    <SortWrap>
      <Select small inline id="nags_sort" label="Sort By" handleChange={e => nagsSort(e.target.value)}>
        <option value="nextNag">Next Nag</option>
        <option value="lastNag">Last Nag</option>
        <option value="AZ">A-Z</option>
        <option value="ZA">Z-A</option>
        <option value="latest">Latest First</option>
        <option value="oldest">Oldest First</option>
      </Select>
    </SortWrap>
  </SearchSortWrap>
);

SearchSort.propTypes = {
  nagsSearch: PropTypes.func.isRequired,
  nagsSort: PropTypes.func.isRequired
}

export default SearchSort;
