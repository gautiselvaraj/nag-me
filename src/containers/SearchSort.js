import { connect } from 'react-redux';
import SearchSort from '../components/SearchSort';
import { nagsSearch, nagsSort } from '../actions/NagActions';

const mapDispatchToProps = dispatch => ({
  nagsSearch: query => dispatch(nagsSearch(query)),
  nagsSort: sortBy => dispatch(nagsSort(sortBy))
});

export default connect(null, mapDispatchToProps)(SearchSort);
