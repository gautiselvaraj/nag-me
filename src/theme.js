export default {
  main:           '#214154',
  mainDarken:     '#083148',
  mainLighten:    '#215a7b',
  black:          '#111111',
  white:          '#f6f6f6',
  greyLightest:   '#efefef',
  greyLighter:    '#dddddd',
  greyLight:      '#a5a5a5',
  grey:           '#898989',
  greyDark:       '#777777',
  greyDarker:     '#595959',
  greyDarkest:    '#222',
}
