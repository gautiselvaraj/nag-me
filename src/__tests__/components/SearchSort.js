import React from 'react';
import { mount } from 'enzyme';
import SearchSort from '../../components/SearchSort';
import theme from '../../theme';
import 'jest-styled-components';

describe('<SearchSort />', () => {
  let nagsSearch, nagsSort;

  beforeEach(() => {
    nagsSearch = jest.fn();
    nagsSort = jest.fn();
  });

  it('should match snapshot and have correct elements & styles', () => {
    let searchSort = mount(
      <SearchSort
        query=""
        nagsSearch={nagsSearch}
        nagsSort={nagsSort}
        theme={theme}
      />
    );
    expect(searchSort).toMatchSnapshot();

    const searchInput = searchSort.find('input[placeholder="Search"]');
    expect(searchInput.length).toBe(1);
    searchInput.simulate('change', { target: { value: 'Search Query' } });
    expect(nagsSearch.mock.calls[0][0]).toBe('Search Query');

    const sortInput = searchSort.find('select#nags_sort');
    expect(sortInput.length).toBe(1);
    sortInput.simulate('change', { target: { value: 'AZ' } });
    expect(nagsSort.mock.calls[0][0]).toBe('AZ');

    expect(searchSort.find('select#nags_sort option').length).toBe(6);
  });
});
