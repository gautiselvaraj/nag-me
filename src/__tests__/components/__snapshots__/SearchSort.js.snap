// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SearchSort /> should match snapshot and have correct elements & styles 1`] = `
.c0 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  border-bottom: 1px solid;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  padding: 5px;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
}

.c1 {
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
}

.c2 {
  position: relative;
}

.c2 input {
  padding-left: 20px;
}

.c2:before {
  content: '\\e986';
  font-family: icons;
  font-size: 0.5rem;
  left: 7px;
  position: absolute;
  top: 7px;
}

.c3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.c4 {
  border: 1px solid;
  border-radius: 5px;
  box-sizing: border-box;
  font-size: .85rem;
  height: 1.5rem;
  padding: 0 0.35rem;
  width: 100%;
}

.c5 {
  margin-left: 5px;
}

.c6 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
}

.c7 {
  font-size: .8rem;
  line-height: 1.5rem;
  margin-right: 5px;
}

.c8 {
  border: 1px solid;
  box-sizing: border-box;
  font-size: .85rem;
  height: 1.5rem;
  margin-top: 0;
  margin-bottom: 0;
}

<SearchSort
  nagsSearch={[Function]}
  nagsSort={[Function]}
  query=""
  theme={
    Object {
      "black": "#111111",
      "grey": "#898989",
      "greyDark": "#777777",
      "greyDarker": "#595959",
      "greyDarkest": "#222",
      "greyLight": "#a5a5a5",
      "greyLighter": "#dddddd",
      "greyLightest": "#efefef",
      "main": "#214154",
      "mainDarken": "#08283B",
      "mainLighten": "#3B5B6E",
      "white": "#f6f6f6",
    }
  }
>
  <styled.div>
    <div
      className="c0"
    >
      <styled.div>
        <div
          className="c1"
        >
          <Component
            name="query"
            onChange={[Function]}
            value=""
          >
            <styled.div>
              <div
                className="c2"
              >
                <Component
                  name="query"
                  onChange={[Function]}
                  placeholder="Search"
                  small={true}
                  type="search"
                  value=""
                >
                  <styled.div
                    name="query"
                    onChange={[Function]}
                    placeholder="Search"
                    small={true}
                    type="search"
                    value=""
                  >
                    <div
                      className="c3"
                      name="query"
                      onChange={[Function]}
                      placeholder="Search"
                      type="search"
                      value=""
                    >
                      <styled.input
                        name="query"
                        onChange={[Function]}
                        placeholder="Search"
                        small={true}
                        type="search"
                        value=""
                      >
                        <input
                          className="c4"
                          name="query"
                          onChange={[Function]}
                          placeholder="Search"
                          type="search"
                          value=""
                        />
                      </styled.input>
                    </div>
                  </styled.div>
                </Component>
              </div>
            </styled.div>
          </Component>
        </div>
      </styled.div>
      <styled.div>
        <div
          className="c5"
        >
          <Component
            id="nags_sort"
            inline={true}
            label="Sort By"
            name="sortBy"
            onChange={[Function]}
            small={true}
            value=""
          >
            <styled.div
              inline={true}
              name="sortBy"
              onChange={[Function]}
              small={true}
              value=""
            >
              <div
                className="c6"
                name="sortBy"
                onChange={[Function]}
                value=""
              >
                <Component
                  htmlFor="nags_sort"
                  inline={true}
                  name="sortBy"
                  onChange={[Function]}
                  small={true}
                  value=""
                >
                  <styled.label
                    htmlFor="nags_sort"
                    inline={true}
                    name="sortBy"
                    onChange={[Function]}
                    small={true}
                    value=""
                  >
                    <label
                      className="c7"
                      htmlFor="nags_sort"
                      name="sortBy"
                      onChange={[Function]}
                      value=""
                    >
                      Sort By
                    </label>
                  </styled.label>
                </Component>
                <styled.select
                  id="nags_sort"
                  inline={true}
                  name="sortBy"
                  onChange={[Function]}
                  small={true}
                  value=""
                >
                  <select
                    className="c8"
                    id="nags_sort"
                    name="sortBy"
                    onChange={[Function]}
                    value=""
                  >
                    <option
                      value="nextNag"
                    >
                      Next Nag
                    </option>
                    <option
                      value="lastNag"
                    >
                      Last Nag
                    </option>
                    <option
                      value="AZ"
                    >
                      A-Z
                    </option>
                    <option
                      value="ZA"
                    >
                      Z-A
                    </option>
                    <option
                      value="latest"
                    >
                      Latest First
                    </option>
                    <option
                      value="oldest"
                    >
                      Oldest First
                    </option>
                  </select>
                </styled.select>
              </div>
            </styled.div>
          </Component>
        </div>
      </styled.div>
    </div>
  </styled.div>
</SearchSort>
`;
